### PR TITLE
Add basic text drawing

### PR DESCRIPTION
**Pull request by forslund**
This is currently just proof of concept but can be used to show the Pairing code.

TODO: is improving speed

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -25,6 +25,61 @@
...
+
+    return font
+
+
+def write_fb(im, dev=None):
+    """ Write Image Object to framebuffer.
+
+        TODO: Check memory mapping
+    """
+    dev = dev or '/dev/fb0'
```
Comment (davidwagnerkc) [inline __init__.py]: How about `dev='/dev/fb0'` as default instead of `None` ?

File: __init__.py [hunk 2]
```diff
@@ -25,6 +25,61 @@
...
+    start_time = time.time()
+    cols = []
+    for j in range(im.size[1] - 1):
+        for i in range(im.size[0]):
+            R, G, B, A = im.getpixel((i, j))
+            # Write color data in the correct order for the screen
+            cols.append(struct.pack('BBBB', B, G, R, A))
+    LOG.info('Row time: {}'.format(time.time() - start_time))
+    with open(dev, 'wb') as f:
+        color = [BACKGROUND[2], BACKGROUND[1], BACKGROUND[0], 0]
```
Comment (davidwagnerkc) [inline __init__.py]: namedtuple instead? BACKGROUND.red seems odd so maybe not.

File: __init__.py [hunk 3]
```diff
@@ -25,6 +25,61 @@
...
+    LOG.info('Draw time: {}'.format(time.time() - start_time))
+
+
+def draw_file(file_path, dev=None):
+    """ Writes a file directly to the framebuff device.
+    Arguments:
+        file_path (str): path to file to be drawn to frame buffer device
+        dev (str): Optional framebuffer device to write to
+    """
+    dev = dev or '/dev/fb0'
```
Comment (davidwagnerkc) [inline __init__.py]: Reason this isn't just function default?

File: __init__.py [hunk 4]
```diff
@@ -25,6 +25,61 @@
...
+    cols = []
+    for j in range(im.size[1] - 1):
+        for i in range(im.size[0]):
+            R, G, B, A = im.getpixel((i, j))
+            # Write color data in the correct order for the screen
+            cols.append(struct.pack('BBBB', B, G, R, A))
+    LOG.info('Row time: {}'.format(time.time() - start_time))
+    with open(dev, 'wb') as f:
+        color = [BACKGROUND[2], BACKGROUND[1], BACKGROUND[0], 0]
+        f.write(struct.pack('BBBB', *color) * ((800 - im.size[1]) // 2  * 400))
```
Comment (davidwagnerkc) [inline __init__.py]: `WIDTH, HEIGHT = 400, 800` constants would be nice

File: __init__.py [hunk 5]
```diff
@@ -76,6 +131,7 @@ def initialize(self):
             # System events
             self.add_event('system.reboot', self.handle_system_reboot)
             self.add_event('system.shutdown', self.handle_system_shutdown)
+            self.add_event('enclosure.mouth.text', self.handle_show_text)
```
Comment (davidwagnerkc) [inline __init__.py]: I think this is fine for now until we figure out if we want to only do special case for pairing for a while or if we want to support other text messages with this event or do our own enclosure. L129 reset_face only resets the display immediately after first display of showing the pairing code so maybe that is no longer needed? Don't need to mess with now, but I commented it out on my unit so it shows a bit better on Friday.

File: __init__.py [hunk 6]
```diff
@@ -102,6 +158,22 @@ def handle_system_reboot(self, message):
...
+            text = text.strip()
+            font = fit_font(text, self.find_resource(FONT_PATH, 'ui'), 30)
+            w, h = font.getsize(text)
+            image = Image.new('RGBA', (400, h), BACKGROUND)
+            draw = ImageDraw.Draw(image)
+            # Draw to center of screen
+            draw.text(((400 - w) / 2, 0), text,
+                      fill='white', font=font)
+            write_fb(image)
+            time.sleep(30)
```
Comment (davidwagnerkc) [inline __init__.py]: Our Mark 2 Pi enclosure will support time parameter in event? Or display_until_event (like mycroft.paired)?

File: __init__.py [hunk 7]
```diff
@@ -168,7 +240,7 @@ def reset_face(self, message):
             Sets switches from resting "face" to a registered resting screen.
         """
         time.sleep(1)
-        self.collect_resting_screens()
+        draw_file(self.find_resource('mycroft.fb', 'ui'))
```
Comment (davidwagnerkc) [inline __init__.py]: Reason for the sleep above?

File: __init__.py (hunk 1 above)
Comment (forslund) [reply __init__.py]: Will do, I'm never sure which type is mutable and which is immutable so I generally set the default to None to be sure :)

File: __init__.py (hunk 6 above)
Comment (forslund) [reply __init__.py]: If we do something like the mark-2 enclosure there would be an idle override for any shown screen. either infinite or a set number of seconds.

File: __init__.py (hunk 7 above)
Comment (forslund) [reply __init__.py]: Left over from the original mark-2 skill. will remove

**Pull request merged**